analyze/analyze.rs

```diff
@@ -213,58 +213,70 @@ impl traits::Emit for DominatorTree {
             (Align::Left, "Dominator Tree".to_string()),
         ]);
 
-        let dominators = &self.dominators;
+        let opts = &self.opts;
+
+        let mut row = 0 as usize;
 
         fn recursive_add_rows(
             table: &mut Table,
             items: &ir::Items,
             dominator_tree: &BTreeMap<ir::Id, Vec<ir::Id>>,
             depth: usize,
-            dominators: &opt::Dominators,
+            mut row: &mut usize,
+            opts: &opt::Dominators,
             id: ir::Id,
         ) {
             assert_eq!(id == items.meta_root(), depth == 0);
 
-            if dominators.max_row != 0 && depth != dominators.max_depth {
-                if depth > 0 {
-                    let item = &items[id];
+            if let Some(max_rows) = opts.max_rows {
```

Comment:
Do we have a better syntax here? 

This [RFC](https://github.com/rust-lang/rfcs/issues/929) should be handy

Reply:
Yeah that would be nice, but it is fine as-is, and better than it was before with the indentation of everything. Thanks again!